api/v1beta1/types_class.go

```diff
@@ -415,7 +415,6 @@ type AzureManagedMachinePoolClassSpec struct {
 
 // ManagedControlPlaneVirtualNetworkClassSpec defines the ManagedControlPlaneVirtualNetwork properties that may be shared across several managed control plane vnets.
 type ManagedControlPlaneVirtualNetworkClassSpec struct {
-	Name      string `json:"name"`
```

Comment:
I saw your previous comment and realized that I just forgot to revert this since we're still keeping the CRDs the same. The bug fix will be in the template webhook which defaulted the vnet to the same name.

Reply:
Case-by-case basis? What's the path that is the easiest to maintain going forward (and easiest for future maintainers to understand the history of how we got here)?

Reply:
I think that it makes a bit more sense to move the name out of the class spec so that the logic is only incorrect in one place (VirtualNetwork in the template type not being set to the respective vnet template type: https://github.com/willie-yao/cluster-api-provider-azure/blob/11e92174a84cd508f5900a68b135016ef75d56b3/api/v1beta1/types_class.go#L97).

Reply:
The name should never be shared, so leaving it in the template spec may be a bit confusing.

Reply:
That sounds like a good rule of thumb. Here it seems like mostly a wash to me.

Reply:
> Here it seems like mostly a wash to me.

Do you mean that we should not change the type here? Sorry I'm just unfamiliar with the saying but I think either way works!

Reply:
I mean I could go either way here.

Reply:
No strong preference

Reply:
@jackfrancis What is your opinion on moving the `name` out of the template type?

Reply:
I don't fully understand the ramifications, let's maybe have a real-time discussion on a public zoom soon so we can figure this out

Reply:
@jackfrancis We discussed this in last week's office hours and the general consensus was to move this out of the template type, but clarify in the release notes that this is a breaking change for Go API but not real-world usage.

Reply:
in which case lgtm